openedx/features/philu_courseware/models.py

```diff
@@ -13,6 +17,63 @@ def revert_user_post_assessment_attempts(self, user, problem_id):
         deleted_records_count = delete_result[0]
         return deleted_records_count
 
+    def get_score(self, user, chapter_id):
+        """
+        Return competency assessments scores of user in chapter
+
+        :param user: user
+        :param chapter_id: chapter url_name.
+        :return: assessments score dictionary
+        """
+
+        pre_assessment_attempted = None
+        pre_assessment_score = post_assessment_score = attempted_pre_assessments = attempted_post_assessments = 0
+
+        query_format = """
+            SELECT MAX(`id`) AS `id`, COUNT(`assessment_type`) AS `assessments_count`, `assessment_type`, `correctness`
+            FROM `philu_courseware_competencyassessmentrecord`
+            WHERE `id` IN (
+                SELECT MAX(`id`) FROM `philu_courseware_competencyassessmentrecord`
+                WHERE `chapter_id` = '{chapter_id}' and `user_id` = {user_id}
+                GROUP BY `problem_id`, `question_number`
+            ) GROUP BY `correctness`, `assessment_type`
+        """
```

Comment:
Wouldn't it be better if this query is placed in a separate file, maybe a constants file. Placing the query in this file violates the rule from the clean code book that we have chosen as a set of our rules:

"It is confusing and sloppy, to have multiple languages in a single source file. Realistically, a source file should contain one, and only one, language."

In this case, we have sql amidst rest of the python code.

Reply:
I don't think sql query strings are something we need to place it in a separate file when django provides managers to write custom or raw queries. As it isn't programing/scripting language but structured which runs by python/django methods.